content/blog/2018/01/2018-01-06-jep-200.adoc

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title: "JEP-200: Remoting / XStream whitelist integrated into Jenkins core"
+tags:
+- core
+- security
+- remoting
+- upgrade
+author: jglick
+---
+
+== Overview
+
+link:https://github.com/jenkinsci/jep/blob/master/jep/200/README.adoc[JEP-200] has been integrated into Jenkins weekly builds
+and (if all goes well) will be a part of the next LTS line.
+In a nutshell, this change is a security hardening measure
+to be less permissive about deserializing Java classes defined in the Java Platform or libraries bundled with Jenkins.
+For several years now, Jenkins has specifically _blacklisted_ certain classes and packages according to known or suspected exploits;
+now it will reject all classes not explicitly mentioned in a _whitelist_, or defined in Jenkins core or plugins.
+
+== For Jenkins administrators
+
+To the extent that advance testing of the impact of this change on popular plugins has been completed,
+most users (and even plugin developers) should not notice any difference.
+If you do encounter a `java.lang.SecurityException: Rejected: some.pkg.and.ClassName` in the Jenkins UI or logs,
+you may have found a case where an unusual plugin, or an unusual usage mode of a common plugin,
+violates the existing whitelist.
+This will be visible in the Jenkins system log as a message from `jenkins.security.ClassFilterImpl` like the following:
+
+----
+some.pkg.and.ClassName in file:/var/lib/jenkins/plugins/some-plugin-name/WEB-INF/lib/some-library-1.2.jar might be dangerous, so rejecting; see https://jenkins.io/redirect/class-filter/
```

Comment:
Both XStream and Remoting patterns should be present in the post

Reply:
The log message is the same.